interface/src/app/analyze/analysis/sampleBin.js

```diff
@@ -77,6 +80,26 @@ function($log, $cacheFactory, $q, Sample, Activity) {
       }
     },
 
+    getSampleGroups: function() {
+      var keys = Object.keys(this.sampleGroups);
+      var samplesByGroup = {};
+      var i;
+
+      // each distinct value in sampleGroups becomes a key in samplesByGroup,
+      // and the keys of sampleGroups are collected in a list within each
+      // corresponding value of samplesByGroup
+      for (i = 0; i < keys.length; i++) {
+        if (samplesByGroup[this.sampleGroups[+keys[i]]]) {
```

Comment:
I guess lines 92-97 can be also written a little more compact like this:
```
if (!samplesByGroup[this.sampleGroups[+keys[i]]]) {
  samplesByGroup[this.sampleGroups[+keys[i]]] = [];
}
samplesByGroup[this.sampleGroups[+keys[i]]].push(+keys[i]);
```
or even:
```
samplesByGroup[this.sampleGroups[+keys[i]]] = samplesByGroup[this.sampleGroups[+keys[i]]] || [];
samplesByGroup[this.sampleGroups[+keys[i]]].push(+keys[i]);
```
Just my random thoughts, you don't have to change if you don't like them.